Profile.js

```diff
@@ -29,39 +29,29 @@ Flint.decorateView('Profile', (Profile) =>
 );
 
 view Profile {
-  const handleLogout = () => {
-    if (view.props.onLogout) {
-      view.props.onLogout();
-    }
-  }
+  prop onLogout = () => {}
+  prop viewer
+
+  const getActive = () => {
+    const credentials = viewer.user.credentials
+    const providers = ['google', 'facebook','twitter', 'github']
+    const active = providers
+      .filter(x => credentials[x])
+      .map(x => ({ type: x, displayName: credentials[x].displayName }))
 
-  const getActiveCredential = () => {
-    const credentials = view.props.viewer.user.credentials;
-    for (const provider of ['google', 'facebook', 'twitter', 'github']) {
-      if (credentials[provider]) {
-        return {
-          type: provider,
-          displayName: credentials[provider].displayName
-        };
-      }
-    }
+    return active.length > 0 ? active[0] : null
   }
 
-  const credentials = getActiveCredential();
+  let active = null
```

Comment:
This looks great! Yes, in general we try to stay away from render functions, preferring to call functions in the actual jsx instead. This will make it easier to do subcomponent updates later(instead of updating the whole component, which is much faster).